### PR TITLE
Introduce Linting

### PR DESCRIPTION
**Pull request by judtinzhang**
- `black`, `isort`, `flake8` dependency upgrades
- Linted files
- Added pre-commit
- Add `coverage` (thought not used at the moment)
- With this PR, highly suggest setting up `pre-commit`

### CODE REVIEW

**Review by JeffersonDing — COMMENTED** [review]
LGTM 🥶

File: Pipfile [hunk 1]
```diff
@@ -19,11 +19,16 @@ passlib = "*"
 jwcrypto = "*"
 pytest-asyncio = "*"
 pytest = "*"
+pre-commit = "*"
+click = "==8.0.4"
 
 [dev-packages]
-isort = "*"
```
Comment (dr-Jess) [inline Pipfile:24]: Is there a reason we don't need isort anymore?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[flake8]
+max-line-length = 100
+exclude = .venv, migrations
+inline-quotes = double
+ignore = E203, W503
+
+[isort]
+default_section = THIRDPARTY
+known_first_party = pennmobile, dining, gsr_booking, user, laundry
```
Comment (vcai122) [inline setup.cfg:9]: we should change these right

File: tests/test_load.py [hunk 3]
```diff
@@ -53,7 +55,7 @@ def make_request():
 
 def run_threads():
     with ThreadPoolExecutor(max_workers=THREADS) as executor:
-        futures = [executor.submit(make_request) for _ in range(NUMBER_OF_REQUESTS)]
+        _ = [executor.submit(make_request) for _ in range(NUMBER_OF_REQUESTS)]
```
Comment (vcai122) [inline tests/test_load.py:58]: lol why not 2 lines, this seems wasteful

**Pull request merged**